src/libretro.c

```diff
@@ -0,0 +1,843 @@
+#include "libretro.h"
+
+#include <stdbool.h>
+#include <time.h>
+#include <string.h>
+#include <math.h>
+
+#include "shared.h"
+#include "sb_types.h"
+#include "gb.h"
+#include "gba.h"
+#include "nds.h"
+
+#define SKYEMU_LIBRETRO_VERSION "0.1.0"
+
+// necessary to resolve a linker error when loaded by retroarch
+bool se_load_bios_file(const char* name, const char* base_path, const char* file_name, uint8_t* data, size_t data_size) {
```

Comment:
Its probably better to make this a callback function in the cores that you can just set to NULL in the retro arch load rom commands. That way the cores are completely decoupled from the frontend.

Reply:
I actually quite like how this ended up, I made some changes to this function (that I have yet pushed), so this function is actually used at this point and I don't know if we should remove it still, in favour of a callback?

Reply:
Lets push your set of changes and go from there!